[Feat]:Home Page and its Functionality

ultronjs opened the pull request:
-Menu UI + Functionality
-Home Page UI
-Add Note Functionality
-Update Note Functionality
-Pinned and UnPinned
-Change the Color of the Notes
-UI for Empty Archive, Label, and Trash Page
There are both commented code +console all over the place.
Commented Code is of API integration that has been commented on now because of the issue in Mockbee.
Console.log is also for the debugging of the same.
-Archive Note API Function
-Archive Note Reducer +Context

ultronjs committed:
[Feat]:Home Page and its Functionality

-Menu UI + Functionality
-Home Page UI
-Add Note Functionality
-Update Note Functionality
-Pinned and UnPinned
-Change Color of the Notes
-UI for Empty Archive, Label and Trash Page
There are both commented code +console all over the place.
Commented COde is of API integration that have been commented now because of the issue mockbee.
Console.log is also for the debugging of the same.
-Archive Note API Function
-Archive Note Reducer +Context

tusharkandpal commented on src/context/notesContext.js:
```diff
@@ -0,0 +1,199 @@
+    };
+    const addNoteToArchive = async (note) => {
+      console.log(note);
+      notesDispatch({ type: "ADD_NOTE", payload: note });
+      archiveNotesDispatch({ type: "REMOVE_NOTE", payload: note._id });
```
why these two are getting dispatch here ? should be after success response.

ultronjs replied:
Right Now i am getting 500 in API.thatwhy bypassing it right now

tusharkandpal commented on src/context/notesContext.js:
```diff
@@ -0,0 +1,199 @@
+      }
+    };   
+    const restoreNoteFromArchiveNote = async (note) => {
+      archiveNotesDispatch({ type: "DELETE_ARCHIVE_NOTE", payload: note._id });
+      notesDispatch({ type: "ADD_NOTE", payload: note });
```
repeated code in line `144` & `145`

ultronjs replied:
commented API dispatch on success as API not working right now getting 500

tusharkandpal commented on src/context/notesContext.js:
```diff
@@ -0,0 +1,199 @@
+
+    }
+    const deleteNoteFromArchiveNote = async (note) => {
+      archiveNotesDispatch({ type: "DELETE_ARCHIVE_NOTE", payload: note });
```
again dispatched in line `165` ?

ultronjs replied:
sorry forgot to comment that code.
Done.

tusharkandpal reviewed (approved):
suggested few changes. liked your use of toast & its types.

ultronjs committed:
commented API dispatch on success

ultronjs committed:
Uncommenting API calls

sushil-mod commented on src/components/ArchiveList.jsx:
```diff
@@ -0,0 +1,37 @@
+    deleteNoteFromArchiveNote,
+  } = useNotes();
+  useEffect(() => {
+    console.log("i m getting triggered");
```
remove console log

ultronjs replied:
Will remove this one in refactoring PR

sushil-mod commented on src/components/ColorPallete.jsx:
```diff
@@ -0,0 +1,29 @@
+import React from 'react'
+
+function ColorPallete({setNote }) {
```
Nice you have made your own color palette

sushil-mod commented on src/components/EditNote.jsx:
```diff
@@ -0,0 +1,96 @@
+    setEditNote((prevState) => ({ ...prevState, archive: !prevState.archive }));
+  };
+  console.log(editNote);
+  // const{children}=parse(content).props
+  // console.log(children)
```
can remove commented code

ultronjs replied:
Done

sushil-mod commented on src/components/EditNote.jsx:
```diff
@@ -0,0 +1,96 @@
+            }} />
+          )}
+          <MdLabelOutline size={25} />
+          {/* <MdLabelOutline size={25} />
+        <BsArchive size={25} />
+        <FiTrash size={25} /> */}
```
here also if it is not in use

sushil-mod reviewed (approved):
just a suggestion .overall code is nice

ultronjs committed:
[fix]:Resolving Git Comment

-Resolving Git Comment
-Uncommented API integration
-Linked SignOut Functionality
-Added Search Bar
